openedx/core/djangoapps/password_policy/forms.py

```diff
@@ -24,9 +25,9 @@ def clean(self):
                 password_policy_compliance.enforce_compliance_on_login(self.user_cache, cleaned_data['password'])
             except password_policy_compliance.NonCompliantPasswordWarning as e:
                 # Allow login, but warn the user that they will be required to reset their password soon.
-                messages.warning(self.request, str(e))
+                messages.warning(self.request, HTML(str(e)))
             except password_policy_compliance.NonCompliantPasswordException as e:
                 # Prevent the login attempt.
-                raise ValidationError(str(e))  # lint-amnesty, pylint: disable=raise-missing-from
+                raise ValidationError(HTML(str(e)))  # lint-amnesty, pylint: disable=raise-missing-from
```

Comment:
Instead of converting the string to HTML should just use the `e` which is already an `HTML` object? ([ref](https://github.com/open-craft/edx-platform/blob/557f58c341710d2d03c2a491e6141276d15823b1/openedx/core/djangoapps/password_policy/compliance.py#L74))

Also, it makes sense for the `NonCompliantPasswordWarning` exception to handle the format/structure of the message. (not sure!)

Reply:
@DubeySandeep The `str` was added as part of a XSS security fix (as far as I can remember), so I'm hesitant to remove it. Keeping it would add an extra layer of security and doesn't really affect the performance much.

Reply:
I see, but earlier it was `str(HTML(str))` and now it's `HTML(str(HTML(str)))`, I think this will again raise the XXS issue (in case it existed before)?

Reply:
The existing `str(HTML)` should take care of escaping problematic HTML tags. So it won't be converted back next time. The same has been done at https://github.com/openedx/edx-platform/blob/557f58c341710d2d03c2a491e6141276d15823b1/openedx/core/djangoapps/user_authn/views/login.py#L200.